clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1516,4 +1527,20 @@ RebalanceProtocol getProtocol() {
     boolean poll(Timer timer) {
         return poll(timer, true);
     }
+
+
+
+    final static class TopicPartitionComparator implements Comparator<TopicPartition>, Serializable {
```

Comment:
Need to implement `Serializable` to make spotbug happy 🤷

Reply:
nit: seems a bit weird to have this in the `ConsumerCoordinator`, would make more sense in a utility class or something related to TopicPartitions -- obviously the actual TopicPartition class would be ideal but unfortunately that would require a KIP, not sure what the second best choice would be but maybe `Topic` or `Utils` or even `CollectionUtils` (under the clients/common package)

Reply:
Good point @ableegoldman ! Actually, there's also another partition comparator in [AbstractStickyAssignor.java](https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java#L1021) . We could make them into util class or `TopicPartitions` class if needed.

Reply:
Ack. Just pushed an update with the refactoring, adding a `Utils` class.